src/main/java/com/internet/shop/dao/ProductDao.java

```diff
@@ -0,0 +1,19 @@
+package com.internet.shop.dao;
+
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.Optional;
+
+public interface ProductDao {
+    Product create(Product product);
+
+    Optional<Product> getById(Long productId);
+
+    Product update(Product product);
+
+    boolean deleteById(Long productId);
+
+    boolean delete(Product product);
```

Comment:
we can remove this method - we won't use it in our app

Reply:
done